### PR TITLE
advanced fitting options support

### PR DESCRIPTION
**Pull request by jeffeaton**
Add support for advanced fitting options:

* Use max_iterations option to specify maximum iterations for optimizer.
* Sed a RNG seed for uncertainty sample simulation so same data --> same outputs.
* `fit_tmb()` gives a warning for convergence errors. `hintr_run_model()` throws an error for convergence errors, unless `option$permissive = TRUE`. 
* Fitting functions are quiet by default.
* Clear some `dplyr` `factor`/`character` join warnings and make `sf::st_write(..., quiet=TRUE)`.
* Restructure `test-run-model.R` to have a global default `data` and `option` set to avoid copy/paste for new tests -> more streamlined updating for added options.
* Inadvertently upgraded to Roxygen 7.0.1 which rewrote much documentation.

This can either be merged into #36 and consider altogether, or merge #36 and then this one.

I am particularly proud of this PR because I wrote tests for the new functionality, so please give some praise for that even if you dislike the rest of it.

Locally this passes `hintr` tests with version on mrc-ide/hintr#93.  I tried to create a separate hintr PR pointing to this branch to run Travis checks (mrc-ide/hintr#94), but think I did not do it right.

### CODE REVIEW

File: inst/metadata/advanced_run_options.json [hunk 1]
```diff
@@ -145,68 +145,67 @@
...
+			    "id": "true",
+			    "label": "Yes"
+			},
+			{
+			    "id": "false",
+			    "label": "No"
+			}
+		    ],
+		    "value": "false",
+		    "helpText": "Select 'Yes' to allow fits with convergence or data errors."
```
[the remark this answers is not in the document]
Comment (r-ash) [reply inst/metadata/advanced_run_options.json]: Perhaps a few of the slides we're presenting can explain these model options and contain this help text.

File: tests/testthat/test-run-model.R [hunk 2]
```diff
@@ -1,35 +1,48 @@
 context("run-model")
 
+## A single set of valid model options and data, update once instead of copying
+## for every test.
+
+a_hintr_data <- list(
```
Comment (r-ash) [inline tests/testthat/test-run-model.R]: Can we stick these in a `setup-*` file?

Comment (jeffeaton) [reply tests/testthat/test-run-model.R]: I've done this in #41 --

**Comment by r-ash** [comment]
Fantastic tests though!

**Pull request merged**